### PR TITLE
feat(core): Improve Langsmith traces for AI executions

### PR DESCRIPTION
**Pull request by OlegIvaniv**
## Summary
This PR improves the tracings that Langchain collects and sends to Lagsmith. 
- Set name of the run in `[$workflow_name] $node_name` format
- Enhance run metadata with execution id, node name, workflow id, whether the workflow is activated, and workflow id
- To track sub-workflow runs(when using Workflow Tool or Workflow Retriever), we need to pass through the CallbackManager of the parent node(`runManager.getChild()`). This is done via `additionalData`, which now accepts the optional `parentRunManager` for these purposes. 

CleanShot 2024-04-08 at 11 06 30

@krynble: I had to add the `@langchain/core` module to the `n8n-workflow` package to set the correct types. I know you had some reservations regarding this in the past, so I'd be open to suggestions on how to handle this. Since we also need the `CallbackManager` type in the `n8n-core` package, I exported it from the `n8n-workflow` so we don't need to also install the LC package there.

## Review / Merge checklist
- [ ] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included.
   > A bug is not considered fixed, unless a test is added to prevent it from happening again.
   > A feature is not complete without tests.

### CODE REVIEW

**Review by krynble — DISMISSED** [review]
LGTM - only checked `core` and `workflow` changes

File: packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/ToolWorkflow.node.ts [hunk 1]
```diff
@@ -384,8 +388,13 @@ export class ToolWorkflow implements INodeType {
 			const items = [newItem] as INodeExecutionData[];
 
 			let receivedData: INodeExecutionData;
+			console.log('About to pass runManager to executeWorkflow', runManager?.runId);
```
Comment (michael-radency) [inline packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/ToolWorkflow.node.ts:391]: is it intentionally left here?

**Review by michael-radency — DISMISSED** [review]
LGTM

File: packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/ToolWorkflow.node.ts (hunk 1 above)
Comment (OlegIvaniv) [reply packages/@n8n/nodes-langchain/nodes/tools/ToolWorkflow/ToolWorkflow.node.ts]: No, by accident, I removed it. Good catch 🙌

**Review by michael-radency — APPROVED** [review]

**Pull request merged**